### DISCUSSION TITLE
Transpose File

### DISCUSSION THREAD

**Discussion by joindevopscloud** [daws-76s/.github#7] [Q&A]
Given a text file file.txt, transpose its content.

You may assume that each row has the same number of columns, and each field is separated by the ' ' character.

**Convert Rows into Columns and Columns into Rows**

Example:

If file.txt has the following content:

```
name age
alice 21
ryan 30
```
Output the following:

```
name alice ryan
age 21 30
```

**Comment by Sravanthich1305** [comment]
cat transpose.sh
#!/bin/bash
input_file="$1"
num_cols=$(head -1 "$input_file" | wc -w)
for i in $(seq 1 "$num_cols"); do
    echo $(cut -d ' ' -f "$i" "$input_file")
done
==========
cat data
name age
tanvi 8
reyan 7
junnu 6
===========
output:
===========
name tanvi reyan junnu
age 8 7 6

**Comment by Bhanuprakash-1995** [comment]
#!/bin/bash
file="file1.txt"
name=()
ages=()
while IFS= read -r line; do
    words=($line)
    name+=(${words[0]})
    ages+=(${words[1]})
done <"$file"

echo "${name[*]}"
echo "${ages[*]}"

**Reply by Bhanuprakash-1995** [reply]
file1.txt content 

name age
alice 21
ryan 30


output:
name alice ryan
age 21 30
